gateway-server/src/main/java/org/apache/knox/gateway/deploy/DeploymentFactory.java

```diff
@@ -260,7 +260,13 @@ private static void addMissingDefaultProviders(Topology topology) {
     // for some tests the number of providers are limited to the classpath of the module
     // and exceptions will be thrown as topologies are deployed even though they will
     // work fine at actual server runtime.
-    if (PROVIDER_CONTRIBUTOR_MAP.get("identity-assertion") != null) {
+
+    // SRM: Only add "Default" provider iff it is found in the provider contributor map
+    // There could be cases where service loader might find other
+    // identity-assertion providers e.g. JWTAuthCodeAsserter
+    if (PROVIDER_CONTRIBUTOR_MAP.get("identity-assertion") != null &&
+            PROVIDER_CONTRIBUTOR_MAP.get("identity-assertion").keySet() != null &&
+            PROVIDER_CONTRIBUTOR_MAP.get("identity-assertion").keySet().contains("Default")) {
```

Comment:
This doesn't seem to be specific to WebShell - is this a generic issue that should be pulled out separately?

Reply:
This was a cause of of NPE when I was working on Webshell. I think I encountered this when working on JWT validation stuff.

Reply:
Done. KNOX-2825